pkg/kwok/controllers/controller.go

```diff
@@ -104,7 +104,13 @@ func NewController(conf Config) (*Controller, error) {
 			return selector.Matches(labels.Set(node.Annotations))
 		}
 	case conf.ManageNodesWithLabelSelector != "":
-		// client-go supports label filtering, so ignore this.
```

Comment:
As this comment says we don't need to use this function to filter node we should check for nil before calling this function😄